oshi-core/src/main/java/oshi/software/os/windows/WindowsNetworkParams.java

```diff
@@ -0,0 +1,136 @@
+import oshi.jna.platform.windows.IPHlpAPI.FIXED_INFO;
+import oshi.jna.platform.windows.Kernel32;
+import oshi.software.common.AbstractNetworkParams;
+import oshi.util.platform.windows.WmiUtil;
+
+public class WindowsNetworkParams extends AbstractNetworkParams {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(WindowsNetworkParams.class);
+
+    private static final WmiUtil.ValueType[] GATEWAY_TYPES = {WmiUtil.ValueType.STRING, WmiUtil.ValueType.UINT16};
+    private static final String IPV4_DEFAULT_DEST = "0.0.0.0/0";
+    private static final String IPV6_DEFAULT_DEST = "::/0";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDomainName() {
+        char[] buffer = new char[256];
+        IntByReference bufferSize = new IntByReference(buffer.length);
+        if (!Kernel32.INSTANCE.GetComputerNameEx(Kernel32.ComputerNameDnsDomain, buffer, bufferSize)) {
+            LOG.error("Failed to get dns domain name. Error code: {}", Kernel32.INSTANCE.GetLastError());
+            return "";
+        }
+        return new String(buffer);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String[] getDnsServers() {
+        // this may be done by iterating WMI instances
+        // ROOT\CIMV2\Win32_NetworkAdapterConfiguration
+        // then sort by IPConnectionMetric, but current JNA release does not
+        // have string array support
+        // for Variant (it's merged but not release yet).
+        WinDef.ULONGByReference bufferSize = new WinDef.ULONGByReference();
+        int ret = IPHlpAPI.INSTANCE.GetNetworkParams(null, bufferSize);
+        if (ret != IPHlpAPI.ERROR_BUFFER_OVERFLOW) {
+            LOG.error("Failed to get network parameters buffer size. Error code: {}", ret);
+            return new String[0];
+        }
+
+        FIXED_INFO buffer = new FIXED_INFO(new Memory(bufferSize.getValue().longValue()));
+        ret = IPHlpAPI.INSTANCE.GetNetworkParams(buffer, bufferSize);
+        if (ret != 0) {
+            LOG.error("Failed to get network parameters. Error code: {}", ret);
+            return new String[0];
+        }
+
+        List<String> list = new ArrayList<>();
+        IPHlpAPI.IP_ADDR_STRING dns = buffer.DnsServerList;
+        while (dns != null) {
+            String addr = new String(dns.IpAddress.String);
+            int nullPos = addr.indexOf(0);
+            if (nullPos != -1) {
+                addr = addr.substring(0, nullPos);
+            }
+            list.add(addr);
+            dns = dns.Next;
+        }
+
+        return list.toArray(new String[list.size()]);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getIpv4DefaultGateway() {
+        return getNextHop(IPV4_DEFAULT_DEST);
+    }
+
+    private String getNextHop(String dest) {
+        Map<String, List<Object>> vals = WmiUtil.selectObjectsFrom("ROOT\\StandardCimv2", "MSFT_NetRoute",
```

Comment:
The `ROOT\\StandardCimv2` requires Windows 8+.  This call produces errors on Windows 7.   

We need to provide a fallback method for finding the default gateway on Win7.  My suggestion would be to parse the output of `netstat -nr` and return the gateway for `0.0.0.0` (IPv4) and `::/0` (IPv6).